pkg/utils/functional/functional.go

```diff
@@ -112,12 +112,12 @@ func InvertStringMap(stringMap map[string]string) map[string]string {
 }
 
 // MaxDuration returns the largest duration
-func MaxDuration(durations ...time.Duration) time.Duration {
-	var max time.Duration
+func MinDuration(durations ...time.Duration) time.Duration {
+	var min time.Duration
 	for _, duration := range durations {
-		if duration > max {
-			max = duration
+		if (duration < min && duration != 0) || (duration != 0 && min == 0) {
```

Comment:
if you initialize min to the first duration, you can simplify this check by removing all the 0s

Reply:
IIUC, if the first duration is 0, we still run into the problem.

Reply:
oh you want to return the smallest non-zero, ok.  I'm guessing it's fine to return 0 if all the durations are 0 then?